### ISSUE TITLE
Bulk Upload error

### ISSUE THREAD

**Issue by gustavnavar** [Microsoft/mattercenter#767]
When I go to the the bulkupload view I get the following error:

Description: Error occured in the system. Please contact the administrator
Message: Exception occured while invoking the endpoint https://graph.microsoft.com/v1.0/me/mailFolders?$skip=0
ErrorCode: 500
MethodName: MakeHttpRequest
ClassName: MailMessageRepository
LineNumber: 274
StackTrace:
System.ApplicationException: Exception occured while invoking the endpoint https://graph.microsoft.com/v1.0/me/mailFolders?$skip=0 ---> System.Net.Http.HttpRequestException: { "error": { "code": "ErrorAccessDenied", "message": "Access is denied. Check credentials and try again.", "innerError": { "request-id": "a56ac115-5117-484a-8cf2-4e471ece5f65", "date": "2017-07-31T14:07:36" } } } --- End of inner exception stack trace --- at Microsoft.Legal.MatterCenter.Repository.MailMessageRepository.MakeHttpRequest[TResult](String httpMethod, String requestUrl, String accept, Object content, String contentType, Func`2 resultPredicate) in C:\Users\Gus\Documents\mc\solution\Microsoft.Legal.MatterCenter.Service\Microsoft.Legal.MatterCenter.Repository\MailMessage\MailMessageRepository.cs:line 274 at Microsoft.Legal.MatterCenter.Repository.MailMessageRepository.MakeGetRequestForString(String graphRequestUrl) in C:\Users\Gus\Documents\mc\solution\Microsoft.Legal.MatterCenter.Service\Microsoft.Legal.MatterCenter.Repository\MailMessage\MailMessageRepository.cs:line 202 at Microsoft.Legal.MatterCenter.Repository.MailMessageRepository.ListFolders(Int32 startIndex) in C:\Users\Gus\Documents\mc\solution\Microsoft.Legal.MatterCenter.Service\Microsoft.Legal.MatterCenter.Repository\MailMessage\MailMessageRepository.cs:line 189 at Microsoft.Legal.MatterCenter.Repository.MailMessageRepository.ListMessages(MailRequest mailRequest) in C:\Users\Gus\Documents\mc\solution\Microsoft.Legal.MatterCenter.Service\Microsoft.Legal.MatterCenter.Repository\MailMessage\MailMessageRepository.cs:line 130 at Microsoft.Legal.MatterCenter.Repository.MailMessageRepository.GetUserInboxEmails(MailRequest mailRequest) in C:\Users\Gus\Documents\mc\solution\Microsoft.Legal.MatterCenter.Service\Microsoft.Legal.MatterCenter.Repository\MailMessage\MailMessageRepository.cs:line 50 at Microsoft.Legal.MatterCenter.Web.EmailController.GetUserInboxEmails(MailRequest mailRequest) in C:\Users\Gus\Documents\mc\solution\Microsoft.Legal.MatterCenter.Web\Controllers\EmailController.cs:line 162 at lambda_method(Closure , Object , Object[] ) at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionFilterAsync>d__28.MoveNext()

It seems that Microsoft.Legal.MatterCenter.Utility.SPOAuthorization.GetTokenForResource(string resourceUrl) is not returning a valid token. I 've also used Postman with the token that I got debugging the app, and I also get a similar error:
{
    "error": {
        "code": "InvalidAuthenticationToken",
        "message": "Access token validation failure.",
        "innerError": {
            "request-id": "b0424308-f63b-48d6-b6ef-2d96d96e7d81",
            "date": "2017-07-31T14:38:11"
        }
    }
}

But if I use the Graph Explorer tool with the same account to call https://graph.microsoft.com/v1.0/me/, I get no error. The same happens if I use Postman with the bearer token the I get with Graph Explorer.

**Comment by premchandpl** [comment]
@gustavnavar In order to use Matter Bulk upload feature, in azure active directory, you need to give permission to Exchange online for Matter Center App. See the below screen shot for the same

![image](https://user-images.githubusercontent.com/3001352/28801342-ebd21b4e-7605-11e7-92a8-8115a2efb224.png)

Do the above change, save the changes and refresh the bulk upload screen and bulk upload functionality should work. Let me know if this solves the issue

**Comment by gustavnavar** [comment]
@premchandpl It did not work, /api/v1/email/getuserinboxemails continues returning an error 500. Any other suggestion?

**Comment by premchandpl** [comment]
@gustavnavar Can you let me know what error you are getting?

**Comment by gustavnavar** [comment]
@premchandpl These are the screenshoots:
![captura de pantalla 2017-07-31 a les 16 09 06](https://user-images.githubusercontent.com/30131949/28834858-d2e3a2a4-76e3-11e7-88db-75bb28d5dd75.png)
![captura de pantalla 2017-07-31 a les 16 10 11](https://user-images.githubusercontent.com/30131949/28834866-d894d5c4-76e3-11e7-9d4b-697f835ff95d.png)

**Comment by gustavnavar** [comment]
@premchandpl  As you pointed out, granting the following permissions to Office 365 Exchange Online:
- Read mail in all mailboxes
- Read user and shared mail
- Read and write user mail
- Read user mail
- Access mailboxes as the signed-in user via Exchange Web Services
and accessing to https://login.microsoftonline.com/tenant/adminconsent?client_id=&state=12345&redirect_uri=https%3a%2f%2flocalhost%2fportorappnumber solved the issue.

After that I found another issue related to the language of the mailbox, in my case in Catalan. The     public class MailMessageRepository.ListMessages(MailRequest mailRequest) throws 2 exceptions because the inbox folder name is "Safata d'entrada". I had to change the line 131 from:
var inboxFolder = folders != null ? folders.Single(folder => folder.Name == "Inbox") : null;
to:
var inboxFolder = folders != null ? folders.SingleOrDefault(folder => folder.Name == "Inbox") : null;
and line 149 from:
messages.MailCount = inboxFolder.TotalItemCount;
to:
messages.MailCount = messages.Messages.Count;
because inboxFolder object is null for a non-English language.

Thanks for you help
